diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9b7934395..e51b34259 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -236,6 +236,18 @@ elseif (CMAKE_SYSTEM_PROCESSOR  STREQUAL "arm64"   OR
   set(C_FPU_X86          OFF)
   set(C_UNALIGNED_MEMORY OFF)
 
+elseif (CMAKE_SYSTEM_PROCESSOR  STREQUAL "powerpc64"   OR
+        CMAKE_SYSTEM_PROCESSOR  STREQUAL "powerpc64le"   OR
+        CMAKE_SYSTEM_PROCESSOR  STREQUAL "ppc64"   OR
+        CMAKE_SYSTEM_PROCESSOR  STREQUAL "ppc64le" OR
+        CMAKE_OSX_ARCHITECTURES STREQUAL "ppc64")
+
+   set(C_TARGETCPU        "PPC64")
+   set(C_DYNAMIC_X86      OFF)
+   set(C_DYNREC           ON)
+   set(C_FPU_X86          OFF)
+   set(C_UNALIGNED_MEMORY ON)
+
 else()
   message(FATAL_ERROR "Unknown processor ${CMAKE_SYSTEM_PROCESSOR}")
 endif()
diff --git a/meson.build b/meson.build
index b195c8fa7..75f4a9d91 100644
--- a/meson.build
+++ b/meson.build
@@ -298,6 +298,14 @@ if is_optimized_buildtype and zlib_ng_is_native
     message('Enabling native SIMD optimizations')
 endif
 
+# ALTIVEC on PowerPC64
+# ~~~~~~~~~~~~~~~
+if target_machine.cpu_family() == 'ppc64'
+    zlib_ng_options += 'altivec'
+    simd_instruction_sets += 'ALTIVEC'
+    message('Using the ALTIVEC instruction set')
+endif
+
 # NEON on Aarch64
 # ~~~~~~~~~~~~~~~
 if target_machine.cpu_family() == 'aarch64'
diff --git a/src/cpu/meson.build b/src/cpu/meson.build
index dd045d53a..dfa2f98b9 100644
--- a/src/cpu/meson.build
+++ b/src/cpu/meson.build
@@ -95,7 +95,7 @@ libcpu = static_library(
         tracy_dep,
         libloguru_dep,
     ],
-    cpp_args: warnings,
+    cpp_args: [ warnings, '-fpermissive' ]
 )
 
 libcpu_dep = declare_dependency(link_with: libcpu)
