--- a/drivers/gpu/drm/radeon/radeon_device.c
+++ b/drivers/gpu/drm/radeon/radeon_device.c
@@ -1366,6 +1366,7 @@ int radeon_device_init(struct radeon_device *rdev,
 	    (rdev->family <= CHIP_RS740))
 		dma_bits = 32;
 #ifdef CONFIG_PPC64
-	if (rdev->family == CHIP_CEDAR)
+	if (rdev->family == CHIP_CEDAR ||
+	    rdev->family == CHIP_CAICOS)
 		dma_bits = 32;
 #endif

--- a/arch/powerpc/kernel/dma-iommu.c
+++ b/arch/powerpc/kernel/dma-iommu.c
@@ -192,23 +192,27 @@ u64 dma_iommu_get_required_mask(struct device *dev)
 u64 dma_iommu_get_required_mask(struct device *dev)
 {
 	struct iommu_table *tbl = get_iommu_table_base(dev);
+	u64 dev_mask;
 	u64 mask;
 
 	if (dev_is_pci(dev)) {
 		u64 bypass_mask = dma_direct_get_required_mask(dev);
 
-		if (dma_iommu_dma_supported(dev, bypass_mask)) {
+		dev_mask = dev->dma_mask ? *dev->dma_mask : 0;
+
+		if (bypass_mask <= dev_mask &&
+		    dma_iommu_dma_supported(dev, bypass_mask)) {
 			dev_info(dev, "%s: returning bypass mask 0x%llx\n", __func__, bypass_mask);
 			return bypass_mask;
 		}
 	}
 
 	if (!tbl)
 		return 0;
 
 	mask = 1ULL << (fls_long(tbl->it_offset + tbl->it_size) +
 			tbl->it_page_shift - 1);
 	mask += mask - 1;
 
 	return mask;
 }
